python_modules/dagster/dagster/core/definitions.py

```diff
@@ -314,7 +409,7 @@ class PipelineDefinition(object):
     '''
 
     def __init__(
-        self, solids, name=None, description=None, context_definitions=None, dependencies=None
+        self, solids=None, name=None, description=None, context_definitions=None, dependencies=None
```

Comment:
let's add a 

`solids = check.opt_list_param(solids, 'solids', SolidDefinition)`

I think this will fatal with solids=None as well unless I am misreading something?

Reply:
Yeah, that's a consequence of double rewrite. Made in non-optional again.